cfme/dashboard.py

```diff
@@ -34,97 +63,246 @@ def reset_widgets(self, cancel=False):
+                else:
+                    return None
+
+            @property
+            def blank(self):
+                return bool(self.browser.elements(self.BLANK_SLATE))
+
+            @classmethod
+            def all(cls, browser):
+                return [(browser.text(e), ) for e in browser.elements(cls.ALL_LOCATOR)]
+
     @property
     def is_displayed(self):
         return (
             self.logged_in_as_current_user and
             self.navigation.currently_selected == ['Cloud Intel', 'Dashboard'])
 
 
-def click_top_right(item):
-    base_locator = '//nav//a[@id="dropdownMenu2"]/../ul//a[normalize-space(.)="{}"]'
-    sel.click(page.user_dropdown)
-    sel.click(base_locator.format(item), wait_ajax=False)
-    sel.handle_alert(wait=False)
+class ParticularDashboardView(DashboardView):
+    @property
+    def is_displayed(self):
+        return (
+            super(ParticularDashboardView, self).is_displayed and
+            self.dashboards(title=self.obj.name).is_active)
 
 
-def click_help(item):
-    base_locator = '//nav//a[@id="dropdownMenu1"]/../ul//a[normalize-space(.)="{}"]'
-    sel.click(page.help_dropdown)
-    sel.click(base_locator.format(item), wait_ajax=False)
-    sel.handle_alert(wait=False)
+class DashboardCollection(Navigatable):
+    @property
+    def default(self):
+        return self.instantiate('Default Dashboard')
 
+    def instantiate(self, name):
+        return Dashboard(name, collection=self)
 
-def reset_widgets(cancel=False):
-    """Resets the widgets on the dashboard page.
+    def all(self):
+        view = navigate_to(self.appliance.server, 'Dashboard')
+        result = []
+        # TODO: Idiomatize the following line
+        for (dashboard_name, ) in view.dashboards.view_class.all(view.browser):
+            result.append(self.instantiate(dashboard_name))
+        return result
+
+    def refresh(self):
+        view = navigate_to(self.appliance.server, 'Dashboard')
+        view.navigation.select('Cloud Intel', 'Dashboard')
+
+    @property
+    def zoomed_name(self):
+        view = navigate_to(self.appliance.server, 'Dashboard')
+        if not view.zoomed.is_displayed:
+            return None
+        return view.zoomed.title.text
+
+    def close_zoom(self):
+        navigate_to(self.appliance.server, 'Dashboard').ensure_zoom_closed()
+
+
+class Dashboard(Navigatable):
+    def __init__(self, name, collection=None, appliance=None):
+        if collection is None:
+            collection = DashboardCollection(appliance=appliance)
+        self.collection = collection
+        Navigatable.__init__(self, appliance=collection.appliance)
+        self.name = name
 
-    Args:
-        cancel: Set whether to accept the popup confirmation box. Defaults to ``False``.
-    """
-    sel.click(page.reset_widgets_button, wait_ajax=False)
-    sel.handle_alert(cancel)
+    @property
+    def dashboard_view(self):
+        return navigate_to(self, 'Details').dashboards(title=self.name)
```

Comment:
@psav Not just specific UI testing, though wil well written views you can perfectly use it for that. Here, because there are two parametrized views nested in each other, one for dashboards and other for widgets, this comes out as a natural use. is_displayed is written well, so it only costs you a second to execute navigate_to and then you get the correct sub-view to work with. And the dashboardwidget will just use this dashboard_view and grab a correct widget subview, making it DRY.